app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaScreen.kt

```diff
@@ -63,6 +70,20 @@ fun ImportMediaScreen(importMediaViewModel: ImportMediaViewModel = hiltViewModel
     ImportMediaContent(importMediaViewModel)
 }
 
+@Composable
+fun checkIfSharingIsEnabled(featureFlagNotificationViewModel: FeatureFlagNotificationViewModel) {
+    featureFlagNotificationViewModel.loadInitialSync()
+
+    val fileSharingRestrictedDialogState = rememberVisibilityState<FileSharingRestrictedDialogState>()
+    FileSharingRestrictedDialogContent(dialogState = fileSharingRestrictedDialogState)
+
+    featureFlagNotificationViewModel.updateSharingStateIfNeeded()
+
+    if (featureFlagNotificationViewModel.featureFlagState.showFileSharingRestrictedDialog) {
+        fileSharingRestrictedDialogState.show(fileSharingRestrictedDialogState.savedState ?: FileSharingRestrictedDialogState)
+    }
+}
+
```

Comment:
I feel like this is something that belongs more to a ViewModel: 

I think this function as a whole could be moved to a Viewmodel I am talking here about the sequence call of :

   featureFlagNotificationViewModel.loadInitialSync() and featureFlagNotificationViewModel.updateSharingStateIfNeeded()

Reply:
I am planning to do some refactoring with this class soon, after I double check the restricted file assets feature is still working seamlessly. But for the sake of unblocking this issue I will do it in a later PR.

Reply:
awesome, a TODO would be nice here